src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,184 @@
+    removePlanes(viewer)
+    const modelCenter1 = new Vector3(0, h1, 0)
+    const modelCenter2 = new Vector3(0, h2, 0)
+    const normal1 = new Vector3(0, 1, 0)
+    const normal2 = new Vector3(0, -1, 0)
+    viewer.clipper.createFromNormalAndCoplanarPoint(normal1, modelCenter1)
+    viewer.clipper.createFromNormalAndCoplanarPoint(normal2, modelCenter2)
+  }
+  const planView = () => {
+    viewer.context.ifcCamera.toggleProjection()
+    viewer.plans.moveCameraTo2DPlanPosition(true)
+    const yConst = 100 // value used in moveCameraTo2DPlanPosition in web-ifc
+    const modelCenterX = getModelCenter(model).x
+    const modelCenterY = getModelCenter(model).y
+    const modelCenterZ = getModelCenter(model).z
+    viewer.context.ifcCamera.cameraControls.setLookAt(modelCenterX, yConst, modelCenterZ, modelCenterX, 0, modelCenterZ, true)
+    const currentProjection = viewer.context.ifcCamera.projectionManager.currentProjection
+    const camFac = 5
+    if (currentProjection === 0) {
+      viewer.context.ifcCamera.cameraControls.setLookAt(
+          modelCenterX * camFac, modelCenterY * camFac, -modelCenterZ * camFac, modelCenterX, modelCenterY, modelCenterZ, true)
+    }
+  }
+  const createPlane = (normalDirection) => {
+    const modelCenter = getModelCenter(model)
+    const planeHash = getHashParams(location, 'p')
+    console.log('in the function modelCenter', modelCenter)
+    let normal
+    switch (normalDirection) {
+      case 'x':
+        normal = new Vector3(1, 0, 0)
+        break
+      case 'y':
+        normal = new Vector3(0, 1, 0)
+        break
+      case 'z':
+        normal = new Vector3(0, 0, 1)
+        break
+      default:
+        normal = new Vector3(0, 1, 0)
+        break
+    }
+    console.log('in the function normal', normal)
+    if (!planeHash || planeHash !== normalDirection ) {
+      addHashParams(window.location, PLANE_PREFIX, {planeAxis: normalDirection})
+    }
+    return viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
+  }
+  useEffect(() => {
+    const planeHash = getHashParams(location, 'p')
+    if (planeHash && model && viewer) {
+      const modelCenter = getModelCenter(model)
+      const planeDirection = planeHash.split(':')[1]
+      let normal
+      switch (planeDirection) {
+        case 'x':
+          normal = new Vector3(1, 0, 0)
+          break
+        case 'y':
+          normal = new Vector3(0, 1, 0)
+          break
+        case 'z':
+          normal = new Vector3(0, 0, 1)
+          break
+        default:
+          normal = new Vector3(0, 1, 0)
+          break
+      }
+      console.log(' ------------------------- ')
+      console.log('the modelCenter from the useEffect', modelCenter)
+      console.log('plane direction', planeDirection)
+      console.log('normal', normal)
+      console.log(' ------------------------- ')
+      createPlane(planeDirection)
+      // viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
```

Comment:
Remove code comments too.  These are sometimes useful, but always try to remove and replace with a doc comment explaining what alternative code could be used.

Reply:
Fixed